### ISSUE TITLE
Converting extracted values (PSLiterals, utf-?)

### ISSUE THREAD

**Issue by typhoon71** [pdfminer/pdfminer.six#182]
First: thanks for pdfminer.six.

Second: my issue is not really with pdfminer, it's that I can't find how to properly do something, so I'm asking for help here, since I think it's already covered in pdfminer.

I'm extracting names+values pairs from some compiled pdf forms using something like this code:

```python
from pdfminer.pdfparser import PDFParser
from pdfminer.pdfdocument import PDFDocument
from pdfminer.pdftypes import resolve1

fp = open(filename, 'rb')
parser = PDFParser(fp)
doc = PDFDocument(parser)
fields = resolve1(doc.catalog['AcroForm'])['Fields']
for i in fields:
    field = resolve1(i)
    name, value = field.get('T'), field.get('V')
```

My problem is that the values I extract are PSLiterals or bytecode (that usually decodes fine as utf8, but sometime it raises UnicodeError), probably because of special or exothic characters.
Sometime it's the ì or à, other “ or ” (from a mac compiled pdf); they're in the \\xc3 form.
In the source pdf form the letter are readable, since they are typed by the user.

Rummaging throug the pdfminer code and tool there seems to be routines that should take care of that, but I can't find examples of how to use them.

I can't share the pdfs, but I can share some sample of the problematic values output if needed.

Thanks.

**Comment by pietermarsman** [comment]
Hi @typhoon71, thanks again for your question! It would be great if you could give some examples of bytecodes that cannot be translated to unicode characters. It would be even better if you also can check how these characters are displayed in a PDF viewer.

**Comment by typhoon71** [comment]
I should still have some problematic pdf/code around, I can post something iin the weekend.

About "how these characters are displayed in a PDF viewer": I'll copy/paste them and make a little png.

I won't be able to post a full pdf, but I'm pretty sure I can post meaningful snippets.

I can investigate the pdf itself if needed (but not super fast).

**Comment by pietermarsman** [comment]
> I should still have some problematic pdfs/code around, I can post something in the weekend.

Great!

**Comment by typhoon71** [comment]
Will do next weekend, some stuff come up and couldn't do anything today.

**Comment by typhoon71** [comment]
I will be busy for the foreseeable future and won't have much time for this.

Still, I found that it's possible to convert from PSLiteral back to string/bytes, which would take care of one of the issues I was having (and it was simpler than expected).
Below it's the updated code that also takes care of resolving value being a PDFObjRef itself on some cases (in the pdf forms I used it on).

``` python
from pdfminer.pdfparser import PDFParser, PSLiteral
from pdfminer.pdfdocument import PDFDocument
from pdfminer.pdftypes import resolve1, PDFObjRef

fp = open(filename, 'rb')
parser = PDFParser(fp)
doc = PDFDocument(parser)
fields = resolve1(doc.catalog['AcroForm'])['Fields']
for i in fields:
    field = resolve1(i)
    name, value = field.get('T'), field.get('V')
    if isinstance(value, PDFObjRef):
        value = resolve1(value)
    if isinstance(value, PSLiteral):
        value = value.name
```
This, depending on the forms, gives value as bytes.

Decoding byte output (ie b'something') should be the next step, the one I got no time for right now.
The valid encodings in the PDF standard are limited, but I want to try to get the encode from the object, whithout going to guess the correct one. Also taking care of unescaping escaped stuff.

Btw, the snippet above doesn't take care of possible exceptions, ie broken PDF structure.
